### PR TITLE
Lab 2 - bubble, insertion and merge sort

### PR DESCRIPTION
**Pull request by loki5265**
Fixes #42

#### Changes proposed in this PR:

* Implement bubble sort
* Implement insertion sort
* Implement merge sort
* Add test case
* Fix code style
* Fix test case
* Fix comments

### CODE REVIEW

File: 02_bubble/lokili/main.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+func insertion(arr []int) []int {
+	arrLen := len(arr)
+	sorted := make([]int, arrLen)
+	copy(sorted, arr)
+
+	for i := 1; i <= arrLen-1; i++ {
+		insertValue := sorted[i]
+		j := i - 1
+		for j >= 0 && sorted[j] > insertValue {
```
Comment (anzdaddy) [inline 02_bubble/lokili/main.go]:
```go
for ; j >= 0 && sorted[j] > insertValue; j-- {
	sorted[j+1] = sorted[j]
}
```

File: 02_bubble/lokili/main.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+var out io.Writer = os.Stdout
+
+func bubble(arr []int) []int {
+	arrLen := len(arr)
+	sorted := make([]int, arrLen)
+	copy(sorted, arr)
+
+	for i := 0; i < arrLen-1; i++ {
+		swapped := false
+		for j := 0; j < arrLen-i-1; j++ {
```
Comment (anzdaddy) [inline 02_bubble/lokili/main.go]: Counting down with `i` simplifies the `j` for-loop:
```go
	for i := arrLen-1; i > 0; i++ {
		swapped := false
		for j := 0; j < i; j++ {
```

File: 02_bubble/lokili/main.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		for j >= 0 && sorted[j] > insertValue {
+			sorted[j+1] = sorted[j]
+			j--
+		}
+		sorted[j+1] = insertValue
+	}
+	return sorted
+}
+
+func mergeSort(arr []int, start int, mid int, end int, tmp []int) {
```
Comment (anzdaddy) [inline 02_bubble/lokili/main.go]:
```go
…, start, mid, end int, …
```

**Review by anzdaddy — CHANGES_REQUESTED** [review]
Have you already encountered slices in this lab?

File: 02_bubble/lokili/main.go (hunk 1 above)
Comment (loki5265) [reply 02_bubble/lokili/main.go]: That's right to simplify the for loop.

File: 02_bubble/lokili/main.go (hunk 2 above)
Comment (loki5265) [reply 02_bubble/lokili/main.go]: I've changed it to the right format.

File: 02_bubble/lokili/main.go (hunk 3 above)
Comment (loki5265) [reply 02_bubble/lokili/main.go]: Sorry for forgetting the go format. Change it.

**Review by anzdaddy — APPROVED** [review]

**Review by juliaogris — APPROVED** [review]

**Pull request merged**